### PR TITLE
chore(test): make snapshot EndToEnd tests hermetic

### PR DESCRIPTION
**Pull request by igorbernstein2**
* zip the test data and store it as a test resource
* update test to unzip and upload the data
* update regen instructions to be a runnable script
* remove old fixture data

### CODE REVIEW

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh [hunk 1]
```diff
@@ -1,4 +1,11 @@
-// Run from HBase shell. Run `hbase shell` from unix terminal on HBase master.
+#!/bin/bash
+
+set -e
+
+# The easiest way to use this is to run it on the master node
+# and then scp /tmp/EndToEndIT-snapshot.zip to the git repo.
+
+hbase shell <<EOF
 create 'test', 'cf', {SPLITS => ["1", "2", "3", "4", "5", "6", "7", "8", "9"]}
```
Comment (vermas2012) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh:9]: Are we invoking this shell script from every test?

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh [hunk 2]
```diff
@@ -102,32 +109,18 @@ put 'test','99', 'cf:a', 'value99', 100
 put 'test','100', 'cf:a', 'value100', 100
 snapshot 'test', 'test-snapshot'
 list_snapshots
+EOF
```
Comment (vermas2012) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh:112]: @TracyCuiCan You need to add the compressed table here. We will also need to replace the zip file with the one with both snapshots.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh [hunk 3]
```diff
@@ -102,32 +109,18 @@ put 'test','99', 'cf:a', 'value99', 100
...
+hbase org.apache.hadoop.hbase.snapshot.ExportSnapshot -snapshot test-snapshot -copy-to /tmp/EndToEndIT-snapshot/data -mappers 16
 
-////////////////////Run from Unix shell on HBase master node//////////////////
-// Export the snapshot
-hbase org.apache.hadoop.hbase.snapshot.ExportSnapshot -snapshot test-snapshot -copy-to /integration-test/data -mappers 16
-
-// Create the hashes for the table. Run the command from unix shell on an HBase
-// node.
-hbase org.apache.hadoop.hbase.mapreduce.HashTable --batchsize=10 --numhashfiles=10 test /integration-test/hashtable
+hbase org.apache.hadoop.hbase.mapreduce.HashTable --batchsize=10 --numhashfiles=10 test /tmp/tmp/EndToEndIT-snapshot/hashtable
```
Comment (vermas2012) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh:118]: `/tmp/tmp`? Is this on purpose?

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java [hunk 4]
```diff
@@ -98,20 +107,20 @@ public void setup() throws Exception {
...
+    fixtureDir = workDir + "hbase-snapshot/";
+    hbaseSnapshotDir = fixtureDir + "data/";
+    hashDir = fixtureDir + "hashtable/";
 
     // Cloud Storage config
     GcpOptions gcpOptions = PipelineOptionsFactory.create().as(GcpOptions.class);
     properties.applyTo(gcpOptions);
     gcsUtil = new GcsUtil.GcsUtilFactory().create(gcpOptions);
 
+    uploadFixture(gcsUtil, SNAPSHOT_FIXTURE_NAME, fixtureDir);
```
Comment (vermas2012) [inline bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java:122]: Should we add a teardown method that cleans up the GCS directory after test is finished?

**Review by vermas2012 — APPROVED** [review]
Overall looks good, Please consider adding cleanup logic for GCS.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java (hunk 4 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java:122]: Yes, but that can happen in a separate pr. I want to keep this focused.

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh (hunk 3 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/resources/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT-snapshot-generate.sh:118]: Oops, that’s a typo

File: bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java (hunk 4 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java:122]: https://github.com/googleapis/java-bigtable-hbase/pull/3609

**Pull request merged**